presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexSupplier.java

```diff
@@ -137,7 +144,16 @@ private static STRtree buildRTree(LongArrayList addresses, List<List<Block>> cha
                 partition = toIntExact(INTEGER.getLong(partitionBlock, blockPosition));
             }
 
-            rtree.insert(getJtsEnvelope(ogcGeometry, radius), new GeometryWithPosition(ogcGeometry, partition, position));
+            GeometryWithPosition geometryWithPosition = new GeometryWithPosition(ogcGeometry, partition, position);
+            rtree.insert(getJtsEnvelope(ogcGeometry, radius), geometryWithPosition);
+
+            estimatedMemorySizeInBytes += ENVELOPE_INSTANCE_SIZE + geometryWithPosition.getEstimatedMemorySizeInBytes();
+
+            // update memory size when estimated change is more than 100 MB
+            if (estimatedMemorySizeInBytes - lastUpdatedMemorySizeInBytes >= 100_000_000) {
+                localUserMemoryContext.setBytes(estimatedMemorySizeInBytes);
```

Comment:
This is dropping the memory used by the PagesIndex itself?

Reply:
Thanks, I included PagesIndex in